src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -638,5 +640,13 @@ public static boolean canCreateFluidSource(World world, BlockPos pos, IBlockStat
         Result result = evt.getResult();
         return result == Result.DEFAULT ? def : result == Result.ALLOW;
     }
+    
+    public static boolean onPistonMove(World world, BlockPos pos, EnumFacing facing, boolean extending)
```

Comment:
I suggest you to make two methods instead of passing a `boolean` argument and check.